examples/dynamo/vgg16_fp8_ptq.py

```diff
@@ -0,0 +1,261 @@
+                if m.bias is not None:
+                    nn.init.constant_(m.bias, 0)
+            elif isinstance(m, nn.BatchNorm2d):
+                nn.init.constant_(m.weight, 1)
+                nn.init.constant_(m.bias, 0)
+            elif isinstance(m, nn.Linear):
+                nn.init.normal_(m.weight, 0, 0.01)
+                nn.init.constant_(m.bias, 0)
+
+    def forward(self, x):
+        x = self.features(x)
+        x = self.avgpool(x)
+        x = torch.flatten(x, 1)
+        x = self.classifier(x)
+        return x
+
+
+def vgg16(num_classes=1000, init_weights=False):
+    vgg16_cfg = [
+        64,
+        64,
+        "pool",
+        128,
+        128,
+        "pool",
+        256,
+        256,
+        256,
+        "pool",
+        512,
+        512,
+        512,
+        "pool",
+        512,
+        512,
+        512,
+        "pool",
+    ]
+    return VGG(vgg16_cfg, num_classes, init_weights)
+
+
+PARSER = argparse.ArgumentParser(
+    description="Load pre-trained VGG model and then tune with FP8 and PTQ"
+)
+PARSER.add_argument("--ckpt", type=str, help="Path to the pre-trained checkpoint")
+PARSER.add_argument(
+    "--batch-size",
+    default=128,
+    type=int,
+    help="Batch size for tuning the model with PTQ",
+)
+PARSER.add_argument(
+    "--fp8-epochs",
+    default=10,
+    type=int,
+    help="The number of epochs to quantize the model to FP8",
+)
+
+args = PARSER.parse_args()
+
+model = vgg16(num_classes=10, init_weights=False)
+model = model.cuda()
+
+# %%
+# Load the pre-trained model weights
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ckpt = torch.load(args.ckpt)
+weights = ckpt["model_state_dict"]
+
+if torch.cuda.device_count() > 1:
+    from collections import OrderedDict
+
+    new_state_dict = OrderedDict()
+    for k, v in weights.items():
+        name = k[7:]  # remove `module.`
+        new_state_dict[name] = v
+    weights = new_state_dict
+
+model.load_state_dict(weights)
+# Don't forget to set the model to evaluation mode!
+model.eval()
+
+# %%
+# Load training dataset and define loss function for PTQ
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+training_dataset = datasets.CIFAR10(
+    root="./data",
+    train=True,
+    download=True,
+    transform=transforms.Compose(
+        [
+            transforms.RandomCrop(32, padding=4),
+            transforms.RandomHorizontalFlip(),
+            transforms.ToTensor(),
+            transforms.Normalize((0.4914, 0.4822, 0.4465), (0.2023, 0.1994, 0.2010)),
+        ]
+    ),
+)
+training_dataloader = torch.utils.data.DataLoader(
+    training_dataset, batch_size=args.batch_size, shuffle=True, num_workers=2
+)
+
+data = iter(training_dataloader)
+images, _ = next(data)
+
+crit = nn.CrossEntropyLoss()
+
+# %%
+# Define Calibration Loop for quantization
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+
+def calibrate_loop(model):
+    # calibrate on a small number of batches
+    for fp8_ep in range(args.fp8_epochs):
```

Comment:
I think this should be only 1 epoch, so we don't need the argument. Calibration only happens once on the dataset (or subset of it), we don't need to iterate it for all epochs